rpc/src/v1/impls/eth.rs

```
@@ -647,8 +655,28 @@ impl<C, S, M, EM> Eth for EthClient<C, S, M, EM> where
 		rpc_unimplemented!()
 	}
 
-	fn compile_solidity(&self, _: Params) -> Result<Value, Error> {
-		try!(self.active());
-		rpc_unimplemented!()
+	fn compile_solidity(&self, params: Params) -> Result<Value, Error> {
+		try!(self.active());
+		from_params::<(String, )>(params)
+			.and_then(|(code, )| {
+				let maybe_child = Command::new("solc")
+					.arg("--bin")
+					.arg("--optimize")
+					.stdin(Stdio::piped())
+					.stdout(Stdio::piped())
+					.stderr(Stdio::null())
+					.spawn();
+				if let Ok(mut child) = maybe_child {
```

Most of the errors are discarded anyway, maybe it would look better using `.and_then` ?